### PR TITLE
use generated files for command line docs

### PR DESCRIPTION
**Pull request by guillermooo**
@FichteFoll
- [ ] ⚠️  rebase before merging ⚠️ 

---

This change is [Reviewable](https://reviewable.io/reviews/guillermooo/sublime-undocs/203)

### CODE REVIEW

File: source/_includes/command_line.g.txt [hunk 1]
```diff
@@ -0,0 +1,84 @@
+.. This file was generated. Don't edit it.
+
+================================== =======================
+``subl [arguments] [files]``       Open given files.
+``subl [arguments] [directories]`` Open given directories.
+``subl [arguments] -``             Edit standard in.
```
Comment (FichteFoll) [inline source/_includes/command_line.g.txt]: All these `[arguments]` should be `[options]`.

**Comment by FichteFoll** [comment]
Reviewed 3 of 3 files at r1.
Review status: all files reviewed at latest revision, 5 unresolved discussions.

---

_[a discussion](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/203#-KIOaij8oHwWYRcHSvAo:-KIOaij8oHwWYRcHSvAp:1076739536) (no related file):_
Are there any notable differences between invoking `subl` or `sublime_text`?

---

_[a discussion](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/203#-KIOap49PHSGyrE2RA2y:-KIOap49PHSGyrE2RA2z:-733339565) (no related file):_
Imo it would be nicer if the argument definitions would use a style similar to Python docstrings, i.e. a brief one-sentence introduction in the first paragraph and then more details in the second and following.

---

_[source/_includes/command_line.g.txt, line 23 [r1]](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/203#-KIO_H4DnQTeW9x5M318:-KIO_H4DnQTeW9x5M319:740406359) ([raw file](https://github.com/guillermooo/sublime-undocs/blob/2962f3f3929e2e5f79ace34197b826d46cc20809/source/_includes/command_line.g.txt#L23)):_

> ```
> =========================== ======================================================
> ``--project`` ``<project>`` Load the given project.
> ``--command`` ``<command>`` Execute the given command.
> ```

These two should be `-- command <command>``(and project) without interrupting the inline code blocks.

---

_[source/_includes/command_line.g.txt, line 42 [r1]](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/203#-KIO_cRDpeoTQBCeyIOF:-KIO_cRDpeoTQBCeyIOG:1354442969) ([raw file](https://github.com/guillermooo/sublime-undocs/blob/2962f3f3929e2e5f79ace34197b826d46cc20809/source/_includes/command_line.g.txt#L42)):_

> ```
>    Execute the given command. The **command** argument
>    designates the command to be run.
>    To pass arguments to the command,
> ```

This section could use a more elaborate description. Like, that `<command>`s are in the form `<command_name> [<args_as_json>]` (probably not like this, but you get the idea). What I mean is that arguments need to be a JSON "object" and separated from the command with a space.

---

_[source/command_line/command_line.rst, line 11 [r1]](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/203#-KIOZenaIIZwGz2SiXEb:-KIOZenaIIZwGz2SiXEc:1789397978) ([raw file](https://github.com/guillermooo/sublime-undocs/blob/2962f3f3929e2e5f79ace34197b826d46cc20809/source/command_line/command_line.rst#L11)):_

> ``` reStructuredText
> 
> Before using ``subl``, make sure
> it is on your ``PATH``.
> ```

We should probably have a sidebar regarding the PATH system variable here.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/203)_

**Comment by FichteFoll** [comment]
Reviewed 2 of 3 files at r1.
Review status: all files reviewed at latest revision, 5 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/203)*

**Comment by FichteFoll** [comment]
Review status: all files reviewed at latest revision, 6 unresolved discussions.

---

_[source/_includes/command_line.g.txt, line 6 [r1]](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/203#-KIOd59GTziBHRbhcGJ3:-KIOd59GTziBHRbhcGJ4:2056723814) ([raw file](https://github.com/guillermooo/sublime-undocs/blob/2962f3f3929e2e5f79ace34197b826d46cc20809/source/_includes/command_line.g.txt#L6)):_

> ```
> ``subl [arguments] [files]``       Open given files.
> ``subl [arguments] [directories]`` Open given directories.
> ``subl [arguments] -``             Edit standard in.
> ```

"in" is not a proper noun. This need to be `standard input` or `stdin`.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/203)_

**Comment by FichteFoll** [comment]
Reviewed 1 of 3 files at r1.
Review status: all files reviewed at latest revision, 6 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/203)*

**Comment by guillermooo** [comment]
Review status: all files reviewed at latest revision, 6 unresolved discussions.

---

_[a discussion](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/203#-KIOaij8oHwWYRcHSvAo:-KIOeFyvFkxl208ukbMe:-1260531473) (no related file):_
<details><summary>

<i>Previously, FichteFoll wrote…</i></summary>



> Are there any notable differences between invoking `subl` or `sublime_text`?
> </details>
> `subl` will parse cmdline arguments using a Windows library. Not sure if there are any differences, though.

---

_[a discussion](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/203#-KIOap49PHSGyrE2RA2y:-KIOeuf0HAOHGx61BhWP:-1170057892) (no related file):_
<details><summary>

<i>Previously, FichteFoll wrote…</i></summary>



> Imo it would be nicer if the argument definitions would use a style similar to Python docstrings, i.e. a brief one-sentence introduction in the first paragraph and then more details in the second and following.
> </details>
> Not sure what you mean. You mean the args to options? Like `<command>` in `--command`?

---

_[source/_includes/command_line.g.txt, line 6 [r1]](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/203#-KIOd59GTziBHRbhcGJ3:-KJoKz0MvWO6skEQKe_O:-499146009) ([raw file](https://github.com/guillermooo/sublime-undocs/blob/2962f3f3929e2e5f79ace34197b826d46cc20809/source/_includes/command_line.g.txt#L6)):_
<details><summary>

<i>Previously, FichteFoll wrote…</i></summary>



> "in" is not a proper noun. This need to be `standard input` or `stdin`.
> </details>
> Done.

---

_[source/_includes/command_line.g.txt, line 23 [r1]](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/203#-KIO_H4DnQTeW9x5M318:-KIOfSeoebbvUrN_EDGR:1834624087) ([raw file](https://github.com/guillermooo/sublime-undocs/blob/2962f3f3929e2e5f79ace34197b826d46cc20809/source/_includes/command_line.g.txt#L23)):_
<details><summary>

<i>Previously, FichteFoll wrote…</i></summary>



> These two should be `-- command <command>``(and project) without interrupting the inline code blocks.
> </details>
> I'll take a look, but that complicates the generator a bit more.

---

_[source/_includes/command_line.g.txt, line 42 [r1]](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/203#-KIO_cRDpeoTQBCeyIOF:-KJoMPuxKkW7oRGB88Ur:-499146009) ([raw file](https://github.com/guillermooo/sublime-undocs/blob/2962f3f3929e2e5f79ace34197b826d46cc20809/source/_includes/command_line.g.txt#L42)):_
<details><summary>

<i>Previously, FichteFoll wrote…</i></summary>



> This section could use a more elaborate description. Like, that `<command>`s are in the form `<command_name> [<args_as_json>]` (probably not like this, but you get the idea). What I mean is that arguments need to be a JSON "object" and separated from the command with a space.
> </details>
> Done.

---

_[source/command_line/command_line.rst, line 11 [r1]](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/203#-KIOZenaIIZwGz2SiXEb:-KJoMSeycvADrr9uyPnU:32851273) ([raw file](https://github.com/guillermooo/sublime-undocs/blob/2962f3f3929e2e5f79ace34197b826d46cc20809/source/command_line/command_line.rst#L11)):_
<details><summary>

<i>Previously, FichteFoll wrote…</i></summary>



> We should probably have a sidebar regarding the PATH system variable here.
> </details>
> Is this required or a nice-to-have?

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/203)_

**Comment by guillermooo** [comment]
To verify:
- [x] can't make `-` (read from stdin) option work on Windows

**Comment by FichteFoll** [comment]
Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---

_[a discussion](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/203#-KIOap49PHSGyrE2RA2y:-KJp7KuC2bbSqNueFQyy:-1175827410) (no related file):_
<details><summary>

<i>Previously, guillermooo (Guillermo López-Anglada) wrote…</i></summary>



> Not sure what you mean. You mean the args to options? Like `<command>` in `--command`?
> </details>
> I meant that as follows (note the new paragraph):

```
``--project <project>``
    Load the given project.

    The **project** argument   designates the :file:`.sublime-project` or :file:`.sublime-workspace` fileto be loaded.
```

---

_[a discussion](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/203#-KIOaij8oHwWYRcHSvAo:-KJp9Yevk0EaUyXZzVcq:1421286301) (no related file):_
<details><summary>

<i>Previously, guillermooo (Guillermo López-Anglada) wrote…</i></summary>



> `subl` will parse cmdline arguments using a Windows library. Not sure if there are any differences, though.
> </details>
> We might as well just propagate to use `subl` to be on the safe side.

---

_[source/_includes/command_line.g.txt, line 6 [r1]](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/203#-KIOd59GTziBHRbhcGJ3:-KJp95QyrAEZzC_GiolA:824230311) ([raw file](https://github.com/guillermooo/sublime-undocs/blob/2962f3f3929e2e5f79ace34197b826d46cc20809/source/_includes/command_line.g.txt#L6)):_
<details><summary>

<i>Previously, guillermooo (Guillermo López-Anglada) wrote…</i></summary>



> Done.
> </details>
> You changed it in line 74 but it seems you missed the table here.

---

_[source/_includes/command_line.g.txt, line 42 [r1]](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/203#-KIO_cRDpeoTQBCeyIOF:-KJp8TVPQkZIU9Pm5P30:1991930364) ([raw file](https://github.com/guillermooo/sublime-undocs/blob/2962f3f3929e2e5f79ace34197b826d46cc20809/source/_includes/command_line.g.txt#L42)):_
<details><summary>

<i>Previously, guillermooo (Guillermo López-Anglada) wrote…</i></summary>



> Done.
> </details>
> Imo the space should be mentioned as well.

---

_[source/_includes/command_line.g.txt, line 42 [r2]](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/203#-KJp8lcu1IRL-tAW5oWF:-KJp8lcu1IRL-tAW5oWG:1325561045) ([raw file](https://github.com/guillermooo/sublime-undocs/blob/8bc07ce28328d63b859a1b0f5d77e0f5b92b49b3/source/_includes/command_line.g.txt#L42)):_

> ```
>    Execute the given command. The **command** argument
>    designates the command to be run.
>    Only ``ApplicationCommand``\s will run
> ```

I'm not 100% sure about this anymore. In some forum thread it was mentioned that window commands could be run as well, but I didn't test this. It could also be that this requires an ST instance to already be open.

If you feel like it, do some tests, otherwise just let it be for the moment and we'll potentially revisit this later.

---

_[source/command_line/command_line.rst, line 11 [r1]](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/203#-KIOZenaIIZwGz2SiXEb:-KJp8f8WJjPXPalzbUkt:-1164104561) ([raw file](https://github.com/guillermooo/sublime-undocs/blob/2962f3f3929e2e5f79ace34197b826d46cc20809/source/command_line/command_line.rst#L11)):_
<details><summary>

<i>Previously, guillermooo (Guillermo López-Anglada) wrote…</i></summary>



> Is this required or a nice-to-have?
> </details>
> nice-to-have

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/203)_

**Comment by guillermooo** [comment]
@fichtefoll Could you test this on your end? I can't make it work.

<details><summary>

<i>Previously, guillermooo (Guillermo López-Anglada) wrote…</i></summary>



> To verify:
> - [x] can't make `-` (read from stdin) option work on Windows
>   </details>

---

Review status: all files reviewed at latest revision, 6 unresolved discussions.

---

_[source/_includes/command_line.g.txt, line 42 [r2]](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/203#-KJp8lcu1IRL-tAW5oWF:-KJpB_CPIUuQq9dhE2dR:-208817708) ([raw file](https://github.com/guillermooo/sublime-undocs/blob/8bc07ce28328d63b859a1b0f5d77e0f5b92b49b3/source/_includes/command_line.g.txt#L42)):_
<details><summary>

<i>Previously, FichteFoll wrote…</i></summary>



> I'm not 100% sure about this anymore. In some forum thread it was mentioned that window commands could be run as well, but I didn't test this. It could also be that this requires an ST instance to already be open.
> 
> If you feel like it, do some tests, otherwise just let it be for the moment and we'll potentially revisit this later.
> </details>
> I'll take a look.

---

_[source/command_line/command_line.rst, line 11 [r1]](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/203#-KIOZenaIIZwGz2SiXEb:-KJpBMFqZT5-wEWD7frB:345137912) ([raw file](https://github.com/guillermooo/sublime-undocs/blob/2962f3f3929e2e5f79ace34197b826d46cc20809/source/command_line/command_line.rst#L11)):_
<details><summary>

<i>Previously, FichteFoll wrote…</i></summary>



> nice-to-have
> </details>
> Then better write it down somewhere :)

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/203#-:-KJpBSAYfyZOWEunIpw5:-741436569)_

**Comment by FichteFoll** [comment]
Me neither, just attempts to open a file named `-`. Does this work on Linux and OSX?

<details><summary>

<i>Previously, guillermooo (Guillermo López-Anglada) wrote…</i></summary>



> @fichtefoll Could you test this on your end? I can't make it work.
> </details>

---

Review status: all files reviewed at latest revision, 6 unresolved discussions.

---

_[source/command_line/command_line.rst, line 11 [r1]](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/203#-KIOZenaIIZwGz2SiXEb:-KJpG3a39b_0QpcQe4Fc:-1383574973) ([raw file](https://github.com/guillermooo/sublime-undocs/blob/2962f3f3929e2e5f79ace34197b826d46cc20809/source/command_line/command_line.rst#L11)):_
<details><summary>

<i>Previously, guillermooo (Guillermo López-Anglada) wrote…</i></summary>



> Then better write it down somewhere :)
> </details>
> https://trello.com/c/zdzQoLEA/109-add-a-sidebar-for-changing-the-system-path-for-cli-page 

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/203#-:-KJpFQDDpB7U-oJ2af4j:-737343967)_

**Comment by guillermooo** [comment]
Only OS X in my tests.

<details><summary>

<i>Previously, FichteFoll wrote…</i></summary>



> Me neither, just attempts to open a file named `-`. Does this work on Linux and OSX?
> </details>

---

Review status: all files reviewed at latest revision, 6 unresolved discussions.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/203#-:-KJpH72cHkMsRUFR6c92:-155892133)_

**Comment by FichteFoll** [comment]
https://github.com/SublimeTextIssues/Core/issues/1236

<details><summary>

<i>Previously, guillermooo (Guillermo López-Anglada) wrote…</i></summary>



> Only OS X in my tests.
> </details>

---

Review status: all files reviewed at latest revision, 6 unresolved discussions.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/203#-:-KJpMY8a6hAZPcgHX9RZ:-353443551)_

**Comment by guillermooo** [comment]
Review status: all files reviewed at latest revision, 6 unresolved discussions.

---

_[a discussion](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/203#-KIOap49PHSGyrE2RA2y:-KJqWPAIQ5P6a8ncSvhb:-499146009) (no related file):_
<details><summary>

<i>Previously, FichteFoll wrote…</i></summary>



> I meant that as follows (note the new paragraph):
> 
> ```
> ``--project <project>``
>     Load the given project.
> 
>     The **project** argument   designates the :file:`.sublime-project` or :file:`.sublime-workspace` fileto be loaded.
> ```
> 
> </details>
> Done.

---

_[source/_includes/command_line.g.txt, line 6 [r1]](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/203#-KIOd59GTziBHRbhcGJ3:-KJqLGOMRgPV0CJkPQLr:-499146009) ([raw file](https://github.com/guillermooo/sublime-undocs/blob/2962f3f3929e2e5f79ace34197b826d46cc20809/source/_includes/command_line.g.txt#L6)):_
<details><summary>

<i>Previously, FichteFoll wrote…</i></summary>



> You changed it in line 74 but it seems you missed the table here.
> </details>
> Done.

---

_[source/_includes/command_line.g.txt, line 42 [r1]](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/203#-KIO_cRDpeoTQBCeyIOF:-KJqWLH4Thgv25ufpUQN:-499146009) ([raw file](https://github.com/guillermooo/sublime-undocs/blob/2962f3f3929e2e5f79ace34197b826d46cc20809/source/_includes/command_line.g.txt#L42)):_
<details><summary>

<i>Previously, FichteFoll wrote…</i></summary>



> Imo the space should be mentioned as well.
> </details>
> Done.

---

_[source/_includes/command_line.g.txt, line 42 [r2]](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/203#-KJp8lcu1IRL-tAW5oWF:-KJqW5lJ06UwWDcHOlAa:-783342773) ([raw file](https://github.com/guillermooo/sublime-undocs/blob/8bc07ce28328d63b859a1b0f5d77e0f5b92b49b3/source/_includes/command_line.g.txt#L42)):_
<details><summary>

<i>Previously, guillermooo (Guillermo López-Anglada) wrote…</i></summary>



> I'll take a look.
> </details>
> Indeed, that seems to be the case. `echo` works only if ST is already running. 

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/203)_

**Comment by FichteFoll** [comment]
Reviewed 1 of 1 files at r3.
Review status: all files reviewed at latest revision, all discussions resolved.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/203)_

**Comment by guillermooo** [comment]
@FichteFoll I meant to squash all changes in this branch :) I think we should make a habit of it.

**Comment by guillermooo** [comment]
Nevermind, tho.

**Comment by FichteFoll** [comment]
I could have done that via github's new merge option, but I merged via reviewable (because I wanted to test it) and that didn't support it.

I say whether we should squash all commits depends on the commits in particular, but it would have been the better choice here.

**Pull request merged**